independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/RequestContext.java

```diff
@@ -126,7 +122,7 @@ public void destroy(Contextual<?> contextual) {
 
     @Override
     public void activate(ContextState initialState) {
-        if (traceEnabled) {
+        if (LOG.isTraceEnabled()) {
             String stack = Arrays.stream(Thread.currentThread().getStackTrace())
```

Comment:
Unrelated but useful: always move away "unlikely" paths ie cold ones out of the "main" method logic: it would improve the change to inline it (see http://normanmaurer.me/blog_in_progress/2013/11/07/Inline-all-the-Things/ that's still very actual and these things hasn't changed across JDK versions, really)

Reply:
Very true, but in this case I preferred the simplest patch to expedite integration & minimize friction for backports.

Definitely something to consider if someone wants to get back to this and improve further.

Reply:
Agree!